### PR TITLE
fix(launcher): `RequestIdStorage` fixes

### PR DESCRIPTION
**Pull request by bragov4ik**
- Fix the layer overwriting fields of handled spans (due to setting `extensions` to some value which are prioritized over any fields)
- Add unit test to check it
- Clippy fixes

### CODE REVIEW

File: libs/blockscout-service-launcher/src/tracing/request_id_layer.rs [hunk 1]
```diff
@@ -77,3 +87,190 @@ impl<S: Subscriber + for<'lookup> LookupSpan<'lookup>> Layer<S> for RequestIdSto
...
+            let span0_with_request_id = tracing::info_span!(
+                "span0_with_request_id",
+                request_id = %"02f09a3f-1624-3b1d-8409-44eff7708208"
+            );
+            let _e0 = span0_with_request_id.enter();
+            let span1 = tracing::info_span!("span1", x = 42);
+            let _e = span1.enter();
+        });
+        let actual = sanitize_timings(make_writer.get_string());
+        assert_eq!(
```
Comment (rimrakhimov) [inline libs/blockscout-service-launcher/src/tracing/request_id_layer.rs:230]: Did the problem exist when `with_span_list(false)` was activated? And shouldn't we test it when all spans are merged instead?

**Review by rimrakhimov — COMMENTED** [review]
Could you please briefly describe how your solution fixes the problem?

File: libs/blockscout-service-launcher/src/tracing/request_id_layer.rs (hunk 1 above)
Comment (bragov4ik) [reply libs/blockscout-service-launcher/src/tracing/request_id_layer.rs:230]: add case with `with_span_list(false)`

**Review by sevenzing — APPROVED** [review]

**Review by lok52 — APPROVED** [review]

**Review by rimrakhimov — APPROVED** [review]

File: libs/blockscout-service-launcher/src/tracing/request_id_layer.rs (hunk 1 above)
Comment (bragov4ik) [reply libs/blockscout-service-launcher/src/tracing/request_id_layer.rs:230]: `with_span_list` doesn't really affect the test and `true` only provides more info to check (just for the record)

**Pull request merged**